sotodlib/site_pipeline/preprocess_tod.py

```diff
@@ -292,18 +343,61 @@ def main(
                 [groups.remove([a[f'dets:{gb}'] for gb in group_by]) for a in x]
                 run_list.append( (obs, groups) )
 
-    logger.info(f"Beginning to run preprocessing on {len(run_list)} observations")
-    for obs, groups in run_list:
-        logger.info(f"Processing obs_id: {obs_id}")
-        try:
-            preprocess_tod(obs["obs_id"], configs, overwrite=overwrite,
-                           group_list=groups, logger=logger)
-        except Exception as e:
-            logger.info(f"{type(e)}: {e}")
-            logger.info(''.join(traceback.format_tb(e.__traceback__)))
-            logger.info(f'Skiping obs:{obs["obs_id"]} and moving to the next')
-            continue
-            
+    # Setup multiprocessing pool.
+    pool = get_context("spawn").Pool(processes=nproc) 
+
+    # Expects archive policy filename to be <path>/<filename>.h5 and then this adds
+    # <path>/<filename>_<xxx>.h5 where xxx is a number that increments up from 0 
+    # whenever the file size exceeds 10 GB.
+    nfile = 0
+    dest_file = os.path.splitext(configs['archive']['policy']['filename'])[0]+'_'+str(nfile).zfill(3)+'.h5'
+    if not(os.path.exists(dest_file)):
+            os.makedirs(dest_file)
```

Comment:
these makedirs are causing me issues because they're making directories and not h5 files. Maybe need to change it to something like 
```
nfile = 0
    dest_file = os.path.splitext(configs['archive']['policy']['filename'])[0]+'_'+str(nfile).zfill(3)+'.h5'
    folder = os.path.split(folder)[0]
    if not (os.path.exists(folder)):
        os.makedirs(folder)
    
    while os.path.exists(dest_file) and os.path.getsize(dest_file) > 10e9:
        nfile += 1
        dest_file = os.path.splitext(dest_file)[0]+str(nfile).zfill(3)+'.h5'
```

both here and below